### ISSUE TITLE
Got a lower accuracy than the original implementation

### ISSUE THREAD

**Issue by ZhenyF** [wandering007/efficient-densenet-pytorch#2]
Hi,
I have traied this model and train it based on this [training code](https://github.com/felixgwu/img_classification_pk_pytorch).

I tried the model using **densenet40 with grouth_rate = 12** on that but only got 6.0% error rate(5.24% inthe report) when using the non-efficiency implementation and 6.1% with efficiency implementation
By the way, the memmory efficiency implementation works pretty good!
Could you please help with that?

**Comment by wandering007** [comment]
@ZhenyF Thanks for the trial. I haven't train densenet on CIFAR10 but I found similar issues when trainining it on CIFAR100. It is mainly due to the subtle difference between Lua Torch and PyTorch, e.g. the different version of cudnn. Maybe cudnn in PyTorch introduces few noises :-) If you really care about the performance, there are two ways I can suggest you to achieve it:  
1. train the model via Lua Torch and port it to PyTorch.  
2. use a modern/sophisticated learning rate schedule (e.g. cyclical learning rates) for better convergence.

BTW, don't wrong about the difference between the efficient and non-efficient versions, they can be treated identically when the experiment settings are the same.

**Comment by ZhenyF** [comment]
Hi @wandering007 
Many thanks for the reply! I have a found out a modification that might improve the accuracy.
The default **esp** in the batchnormalisation2d is 1e-5 in Pytorch while is 1e-3 in Tensorflow.
I modified the esp to 1e-3 and receive a test error rate 5.54% I am not sure if the improved accuracy is caused by that, because I only try it once.(May be I just get a lucky test?)

**Comment by wandering007** [comment]
@ZhenyF That's mystical :-)
